internal/pkg/db/db_revised_test.go

```diff
@@ -0,0 +1,121 @@
+// -----------------------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 bomctl a Series of LF Projects, LLC
+// SPDX-FileName: internal/pkg/db/db_revised_test.go
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// -----------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+package db_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/bomctl/bomctl/internal/pkg/db"
+	"github.com/bomctl/bomctl/internal/testutil"
+)
+
+type dbrSuite struct {
+	suite.Suite
+	*db.Backend
+	documentInfo []testutil.DocumentInfo
+}
```

Comment:
Something about `db_revised_test.go` and `dbrSuite` is nagging at me, hard to pin down. I think the word `revised` is throwing me off, like what is revised referring to, the database? (I know it's referring to a revised document, just the semantics are a little unclear)

To me, it makes more sense to move all of this into `db_test.go`. There are a few things you can try to compact the code and minimize redundancy here and there, such as embedding `dbSuite` into `dbrSuite`:

```go
type (
	dbSuite struct {
		suite.Suite
		*db.Backend
		documents    []*sbom.Document
		documentInfo []testutil.DocumentInfo
	}

	dbRevSuite struct {
		dbSuite
	}

	dbStdSuite struct {
		dbSuite
	}
)
```

`dbSuite` isn't actually run with `suite.Run(t, new(dbSuite))`, it's just to define the common fields and any shared setup/teardown methods you want to implement (`BeforeTest`, `SetupTestSuite`, etc.)

You can probably call both sub-suites from one test function (or separate test functions if you prefer)

```go
func TestDBSuite(t *testing.T) {
	t.Parallel()
	suite.Run(t, new(dbStdSuite))
	suite.Run(t, new(dbRevSuite))
}
```

What are your thoughts?

Reply:
Be careful with embedding though, I just realized I made a mistake embedding `gitClientSuite` into `gitPushSuite` and `gitFetchSuite`, it's running the parse tests defined on the client suite as if all 3 suites had defined them independently 🤦

Reply:
Consolidated them into db_test.go, lmk what you think

Reply:
Looks great!